docs/src/index.md

```diff
@@ -1,37 +1,41 @@
 # DensityInterface.jl
 
-This package defines an interface for mathematical/statistical densities in Julia. The interface comprises the functions [`isdensitytype`](@ref),  [`logdensityof`](@ref) and [`logfuncdensity`](@ref).
+```@docs
+DensityInterface
+```
+
+This package defines an interface for mathematical/statistical densities and objects associated with a density in Julia. The interface comprises the functions [`hasdensity`](@ref),  [`logdensityof`](@ref)/[`densityof`](@ref) and [`logfuncdensity`](@ref).
 
-A the following methods must be provided to make a type (e.g. `SomeDensity`) compatible with the interface:
+The following methods must be provided to make a type (e.g. `SomeDensity`) compatible with the interface:
 
 ```julia
 import DensityInterface
 
-DensityInterface.isdensitytype(::Type{<:SomeType}) = true
-DensityInterface.logdensityof(density::SomeDensity, x) = log_of_density_at_x
+DensityInterface.hasdensity(::SomeDensity) = true
+DensityInterface.logdensityof(d::SomeDensity, x) = log_of_d_at_x
 ```
 
-DensityInterface includes a default implementation of `logdensityof(density)`. It provides a convenient way of passing a log-density function to algorithms like optimizers, samplers, etc.:
+The object `d` may be a density itself or something that can be said to have a density. If `d` is a distribution, the density is the PDF. If `d` is a measure in general, it's density is implied here to be the Radon–Nikodym derivative of `d` and it's base measure. In statistical inference applications, for example, `d` might be a likelihood, prior or posterior.
```

Comment:
I think one could make it clearer that a probability distribution is a probability measure. Currently, it seems that they are different things but the only difference really is the mathematical interpretation/reasoning of the density in the measure theoretical sense. Maybe one could just write:

```suggestion
The object `d` may be a density itself or something that can be said to have a density. If `d` is a distribution, the density is its probability density function. In the measure theoretical sense, the density function is the Radon–Nikodym derivative of `d` with respect to a base measure. In statistical inference applications, for example, `d` - might be a likelihood, prior or posterior.
```

An additional suggestion would be to move the measuretheoretical part to a footnote. This would be less confusing for readers not familiar with measure theory.